nx-X11/programs/Xserver/hw/nxagent/Clipboard.c

```diff
@@ -2590,6 +2639,55 @@ int nxagentConvertSelection(ClientPtr client, WindowPtr pWin, Atom selection,
               NameForIntAtom(target));
   #endif
 
+  if (!nxagentOption(TextClipboard))
+  {
+    /* Optimization: if we have a current target cache check if the
+     * requested target is supported by the owner. If not we can take
+     * a shortcut and deny the request immediately without doing any
+     * further communication */
+    if (targetCache[index].type == FORINT && targetCache[index].forInt)
+    {
+      Atom *targets = targetCache[index].forInt;
+
+      #ifdef DEBUG
+      fprintf(stderr, "%s: Checking target validity\n", __func__);
+      #endif
+      Bool match = False;
+      for (int i = 0; i < targetCache[index].numTargets; i++)
+      {
+        if (targets[i] == target)
+        {
+          match = True;
+          break;
+        }
+      }
+      if (!match)
+      {
+        #ifdef DEBUG
+        fprintf(stderr, "%s: target [%d][%s] is not supported by the owner - denying request.\n",
+                    __func__, target, NameForIntAtom(target));
+        #endif
+        sendSelectionNotifyEventToClient(client, time, requestor, selection, target, None);
+        return 1;
+      }
+    }
```

Comment:
Maybe move this code passage into a function, because it appears twice in a very similar way in this commit (a0599f45e969704ec0edaaac81caaceb57f4a68b).

Reply:
funny note: I thought the same yesterday. But It was too late in the evening...

Reply:
Is not possible because it uses different variable types (XlibAtom vs Atom) at the two locations.